### PR TITLE
Market quote calculations for bond and bill

### PR DESCRIPTION
**Pull request by yukiiwashita**
* Prerequisite for bond curve calibrator
* Added price sensitivity method for bill
* Public method for calculating settlement date, accessing underlying product pricers - these will be used when pricing a trade created from a curve node
* First order derivatives for price-yield conversion

### CODE REVIEW

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricer.java [hunk 1]
```diff
@@ -236,6 +247,20 @@ public CurrencyAmount currentCash(ResolvedBillTrade trade, LocalDate valuationDa
     return CurrencyAmount.zero(trade.getProduct().getCurrency());
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the settlement date.
```
Comment (marc-henrard) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricer.java:252]: Add that in absence of settlement date (position?), the valuation date is returned.

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricer.java [hunk 2]
```diff
@@ -424,8 +435,14 @@ public Payment upfrontPayment(ResolvedFixedCouponBondTrade trade) {
   }
 
   //-------------------------------------------------------------------------
-  // calculate the settlement date
-  private LocalDate settlementDate(ResolvedFixedCouponBondTrade trade, LocalDate valuationDate) {
+  /**
+   * Calculates the settlement date.
```
Comment (marc-henrard) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricer.java:439]: Add that in absence of settlement date (position?), the valuation date is returned.

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricerTest.java [hunk 3]
```diff
@@ -142,8 +142,8 @@ public class DiscountingBillTradePricerTest {
       .build();
   
   // pricers
-  private static final DiscountingBillProductPricer PRICER_PRODUCT = DiscountingBillProductPricer.DEFAULT;
   private static final DiscountingBillTradePricer PRICER_TRADE = DiscountingBillTradePricer.DEFAULT;
+  private static final DiscountingBillProductPricer PRICER_PRODUCT = PRICER_TRADE.getProductPricer();
```
Comment (marc-henrard) [inline modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricerTest.java:146]: Add a test for the new "settlementDate" method.

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricerTest.java [hunk 4]
```diff
@@ -89,8 +89,7 @@ public Payment upfrontPayment(ResolvedFixedCouponBondTrade trade) {
           return Payment.of(CurrencyAmount.zero(trade.getProduct().getCurrency()), SETTLEMENT);
         }
       };
-  private static final DiscountingFixedCouponBondProductPricer PRODUCT_PRICER =
-      DiscountingFixedCouponBondProductPricer.DEFAULT;
+  private static final DiscountingFixedCouponBondProductPricer PRODUCT_PRICER = TRADE_PRICER.getProductPricer();
   private static final DiscountingPaymentPricer PRICER_NOMINAL = DiscountingPaymentPricer.DEFAULT;
   private static final DiscountingFixedCouponBondPaymentPeriodPricer COUPON_PRICER =
       DiscountingFixedCouponBondPaymentPeriodPricer.DEFAULT;
```
Comment (marc-henrard) [inline modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricerTest.java:95]: Add a test for the new "settlementDate" method.

File: modules/data/src/main/java/com/opengamma/strata/data/FieldName.java [hunk 5]
```diff
@@ -57,14 +57,27 @@
    * The field name for the volatility of an asset.
    */
   public static final FieldName VOLATILITY = of("Volatility");
+  /**
+   * The field name for the yield to maturity.
+   * <p>
+   * This is used to refer to the yield of a coupon bond or bill.
+   */
+  public static final FieldName YIELD_TO_MATURITY = of("YieldToMaturity");
```
Comment (marc-henrard) [inline modules/data/src/main/java/com/opengamma/strata/data/FieldName.java:65]: Where is this used in the code?

**Review by marc-henrard — CHANGES_REQUESTED** [review]
Small comments on doc and a missing (trivial) test.

File: modules/data/src/main/java/com/opengamma/strata/data/FieldName.java (hunk 5 above)
Comment (yukiiwashita) [reply modules/data/src/main/java/com/opengamma/strata/data/FieldName.java:65]: Added unit tests

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricer.java (hunk 1 above)
Comment (yukiiwashita) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricer.java:256]: Javadoc updated

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricer.java (hunk 2 above)
Comment (yukiiwashita) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricer.java:454]: Javadoc updated

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricerTest.java (hunk 3 above)
Comment (yukiiwashita) [reply modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricerTest.java:146]: Added tests

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricerTest.java (hunk 4 above)
Comment (yukiiwashita) [reply modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingFixedCouponBondTradePricerTest.java:98]: Added tests

**Review by marc-henrard — APPROVED** [review]
Good for me.

**Pull request merged**